metrics/firebase/test/firestore/rules/builds-test.js

```diff
@@ -141,6 +271,94 @@ describe("", async () => {
+              });
+              let publicDashboardState = config[featureConfigPath].isPublicDashboardEnabled
+                ? 'on'
+                : 'off';
+              let canCreateDescription = user.public_dashboard[publicDashboardState].can.create ?
+                "allows creating a build" : "does not allow creating a build";
+              let canReadDescription = user.public_dashboard[publicDashboardState].can.read ?
+                "allows reading builds" : "does not allow reading builds";
+              let canUpdateDescription = user.public_dashboard[publicDashboardState].can.update ?
+                "allows updating a build" : "does not allow updating a build";
+              let canDeleteDescription = user.public_dashboard[publicDashboardState].can.delete ?
+                "allows deleting a build" : "does not allow deleting a build";
+
+              it(canCreateDescription, async () => {
+                const createPromise = user.app.collection(collection).add(getBuild());
+
+                if (user.public_dashboard[publicDashboardState].can.create) {
+                  await assertSucceeds(createPromise)
+                } else {
+                  await assertFails(createPromise)
+                }
+              });
+
+              it(canReadDescription, async () => {
+                const readPromise = user.app.collection(collection).get();
+
+                if (user.public_dashboard[publicDashboardState].can.read) {
+                  await assertSucceeds(readPromise)
+                } else {
+                  await assertFails(readPromise)
+                }
+              });
+
+              it(canUpdateDescription, async () => {
+                const updatePromise =
+                  user.app.collection(collection).doc("1").update({url: "updated"});
+
+                if (user.public_dashboard[publicDashboardState].can.update) {
+                  await assertSucceeds(updatePromise)
+                } else {
+                  await assertFails(updatePromise)
+                }
+              });
+
+              it(canDeleteDescription, async () => {
+                const deletePromise =
+                  user.app.collection(collection).doc("1").delete();
+
+                if (user.public_dashboard[publicDashboardState].can.delete) {
+                  await assertSucceeds(deletePromise)
+                } else {
+                  await assertFails(deletePromise)
+                }
+              });
+            });
+            callback();
+          });
+        });
+      callback();
+    });
+  });
+
+  after(async () => {
+    await tearDown();
+  });
+});
+
+// Runs general security rules tests.
+describe("", async function () {
+  const passwordProviderAllowedEmailApp = await getApplicationWith(
+    getAllowedEmailUser(passwordSignInProviderId, true)
+  );
+  const collection = "build";
```

Comment:
Let's avoid collection duplication in this file. Please also apply fix in other tests (could be a separate PR for already merged tests).